spec/development/workload/PageIterator.ts

```diff
@@ -0,0 +1,56 @@
+import { PageIterator, PageIteratorCallback } from "../../../src/tasks/PageIterator";
+import { Event } from "microsoft-graph";
+import { assert } from "chai";
+import { getClient } from "../test-helper";
+
+const client = getClient();
+describe("PageIterator", function() {
+	describe("sameHeadersPassedWithPageIterator", () => {
+		it("verify same headers", async () => {
+			const response = await client
+				.api(`/me/events?$top=2`)
+				.headers({ Prefer: 'outlook.timezone= "pacific standard time"' })
+				.select("id,start,end")
+				.get();
+
+			const callback: PageIteratorCallback = (eventResponse) => {
+				const event = eventResponse as Event;
+				assert.equal(event.start.timeZone, "Pacific Standard Time");
+				return true;
+			};
+			var requestOptions = { headers: { Prefer: 'outlook.timezone= "pacific standard time"' } };
+			if (response["@odata.nextLink"]) {
+				const pageIterator = new PageIterator(client, response, callback, requestOptions);
+				await pageIterator.iterate();
+				assert.isTrue(pageIterator.isComplete());
+			}
+		});
+
+		it("differentHeadersPassedWithPageIterator", async () => {
+			const response = await client
+				.api(`/me/events?$top=4`)
+				.headers({ Prefer: 'outlook.timezone= "UTC"' })
+				.select("id,start,end")
+				.get();
+
+			let counter = 0;
+			const callback: PageIteratorCallback = (eventResponse) => {
+				const event = eventResponse as Event;
+				if (counter < 4) {
+					assert.equal(event.start.timeZone, "UTC");
+					counter++;
+				} else {
+					assert.equal(event.start.timeZone, "Pacific Standard Time");
```

Comment:
I'm curious: why would the 4th event and above change timezone?

Reply:
The current implementation is such that if I request for the the top four events with PST timezone, the PST timezone will be applied only to the top four events while the events processed using the PageIterator will have a default timezone as the headers do not get passed in the subsequent requests. Added a test  checking the if the headers passed with the PageIterator are getting applied properly.

Reply:
oh thanks for the explanation! I can see how that could easily trick people, should we potentially have the header stick to the iterator so we can transparently keep sending it? (and have a consistent result across pages)